lux/core/sqltable.py

```diff
@@ -0,0 +1,192 @@
+#  Copyright 2019-2020 The Lux Authors.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pandas as pd
+from lux.core.series import LuxSeries
+from lux.vis.Clause import Clause
+from lux.vis.Vis import Vis
+from lux.vis.VisList import VisList
+from lux.history.history import History
+from lux.utils.date_utils import is_datetime_series
+from lux.utils.message import Message
+from lux.utils.utils import check_import_lux_widget
+from typing import Dict, Union, List, Callable
+
+# from lux.executor.Executor import *
+import warnings
+import traceback
+import lux
+
+
+class LuxSQLTable(lux.LuxDataFrame):
+    """
+    A subclass of pd.DataFrame that supports all dataframe operations while housing other variables and functions for generating visual recommendations.
+    """
+
+    # MUST register here for new properties!!
+    _metadata = [
+        "_intent",
+        "_inferred_intent",
+        "_data_type",
+        "unique_values",
+        "cardinality",
+        "_rec_info",
+        "_min_max",
+        "_current_vis",
+        "_widget",
+        "_recommendation",
+        "_prev",
+        "_history",
+        "_saved_export",
+        "_sampled",
+        "_toggle_pandas_display",
+        "_message",
+        "_pandas_only",
+        "pre_aggregated",
+        "_type_override",
+    ]
+
+    def __init__(self, *args, table_name="", **kw):
+        self._history = History()
+        self._intent = []
+        self._inferred_intent = []
+        self._recommendation = {}
+        self._saved_export = None
+        self._current_vis = []
+        self._prev = None
+        self._widget = None
+        super(LuxSQLTable, self).__init__(*args, **kw)
+        from lux.executor.SQLExecutor import SQLExecutor
+
+        lux.config.executor = SQLExecutor()
+
+        self._sampled = None
+        self._toggle_pandas_display = True
+        self._message = Message()
+        self._pandas_only = False
+        # Metadata
+        self._data_type = {}
+        self.unique_values = None
+        self.cardinality = None
+        self._min_max = None
+        self.pre_aggregated = None
+        self._type_override = {}
+
+        if table_name != "":
+            self.set_SQL_table(table_name)
+        warnings.formatwarning = lux.warning_format
+
```

Comment:
Why do we have a separate `init` function. It looks like `frame.py` takes care of both `SQL` case and `Pandas` case, so why can't we call `super()` and then change whatever we need to?

Reply:
Oh yeah looks like the `init` function in `frame.py` still has the set `table_name` initialization in it. We can actually remove that now that we have a separate LuxSQLTable object to handle the database use case.

Other than that you're right, we can totally just call `super()` and just have the `table_name` set up. Will change this